lib/facter/framework/core/file_loader.rb

```diff
@@ -15,7 +15,10 @@ def load_dir(*dirs)
   return unless Dir.exist?(folder_path.tr('*', ''))
 
   files_to_require = Dir.glob(File.join(folder_path, '*.rb')).reject { |file| file =~ %r{/ffi/} }
-  files_to_require.each(&method(:require))
+  files_to_require.each do |file|
+    rpath = file[File.dirname(__FILE__).length+1..-1]
+    require_relative rpath
```

Comment:
Probably a better way to do this using pathname's `relative_path_from` or at least compute the prefix length once.